model/manifest_test.go

```diff
@@ -17,6 +17,164 @@ import (
+						Type:        "dropdown",
+						HelpText:    "thehelptext",
+						Options: []*PluginOption{
+							{
+								DisplayName: "theoptiondisplayname",
+								Value:       "thevalue",
+							},
+						},
+						Default: "thedefault",
+					},
+				},
+			},
+		}, false},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Title, func(t *testing.T) {
+			err := tc.manifest.IsValid()
+			if tc.ExpectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestIsValidSettingsSchema(t *testing.T) {
+	testCases := []struct {
+		Title          string
+		settingsSchema *PluginSettingsSchema
+		ExpectError    bool
+	}{
+		{"Nil Schema", nil, false},
+		{"Invalid Setting", &PluginSettingsSchema{Settings: []*PluginSetting{{Type: "invalid"}}}, true},
+		{"Happy case", &PluginSettingsSchema{Settings: []*PluginSetting{{Type: "text"}}}, false},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Title, func(t *testing.T) {
+			err := isValidSettingsSchema(tc.settingsSchema)
+			if tc.ExpectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestIsValidSetting(t *testing.T) {
+	testCases := []struct {
+		Title       string
+		setting     *PluginSetting
+		ExpectError bool
+	}{
+		{"Nil setting", nil, false},
+		{"Invalid setting type", &PluginSetting{Type: "invalid"}, true},
+		{"RegenerateHelpText error", &PluginSetting{Type: "text", RegenerateHelpText: "some text"}, true},
+		{"Placeholder error", &PluginSetting{Type: "bool", Placeholder: "some text"}, true},
+		{"Nil Options", &PluginSetting{Type: "bool"}, false},
+		{"Options error", &PluginSetting{Type: "generated", Options: []*PluginOption{}}, true},
+		{"Options displayName error", &PluginSetting{Type: "radio", Options: []*PluginOption{
+			{
+				Value: "some value",
+			},
+		}}, true},
+		{"Options value error", &PluginSetting{Type: "radio", Options: []*PluginOption{
+			{
+				DisplayName: "some name",
+			},
+		}}, true},
+		{"Happy case", &PluginSetting{Type: "radio", Options: []*PluginOption{
+			{
+				DisplayName: "Name",
+				Value:       "value",
+			},
+		}}, false},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Title, func(t *testing.T) {
+			err := isValidSetting(tc.setting)
+			if tc.ExpectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestConvertTypeToPluginSettingType(t *testing.T) {
+	testCases := []struct {
+		Title               string
+		Type                string
+		ExpectedSettingType PluginSettingType
+		ExpectError         bool
+	}{
+		{"bool", "bool", Bool, false},
+		{"dropdown", "dropdown", Dropdown, false},
+		{"generated", "generated", Generated, false},
+		{"radio", "radio", Radio, false},
+		{"text", "text", Text, false},
+		{"longtext", "longtext", LongText, false},
+		{"username", "username", Username, false},
+		{"custom", "custom", Custom, false},
+		{"invalid", "invalid", Custom, true},
```

Comment:
Why does this return `Custom`?

Reply:
I dont check the setting type on error. It would return the first enum otherwise. Updated to Bool.